### ISSUE TITLE
Panic serving api request during Okta authentication integration

### ISSUE THREAD

**Issue by dnoland1** [rancher/rancher#30710]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Unknown, but error happened when attempting to do authentication integration with Okta.

**Result:**
Following pops up in the rancher pod logs:
```
2021/01/06 01:30:08 [ERROR] Panic serving api request:
goroutine 13262 [running]:
runtime/debug.Stack(0xc011bf9dc0, 0x3c26060, 0x7390960)
   /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/rancher/norman/api.(*Server).ServeHTTP.func1(0x4e78ba0, 0xc02df374f0)
   /go/pkg/mod/github.com/rancher/norman@v0.0.0-20200930000340-693d65aaffe3/api/server.go:175 +0x7a
panic(0x3c26060, 0x7390960)
   /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/rancher/rancher/pkg/auth/providers/saml.PerformSamlLogin(0x44d8204, 0x4, 0xc006d85440, 0x3c5dd00, 0xc02de4f4c0, 0x0, 0x0)
   /go/src/github.com/rancher/rancher/pkg/auth/providers/saml/saml_provider.go:118 +0xc7
github.com/rancher/rancher/pkg/auth/providers/publicapi.(*loginHandler).createLoginToken(0xc00bec7ac0, 0xc006d85440, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
   /go/src/github.com/rancher/rancher/pkg/auth/providers/publicapi/login.go:172 +0x905
github.com/rancher/rancher/pkg/auth/providers/publicapi.(*loginHandler).login(0xc00bec7ac0, 0xc0037e28ea, 0x5, 0xc01304dd40, 0xc006d85440, 0xc00bb6d4a0, 0xc00e93f680)
   /go/src/github.com/rancher/rancher/pkg/auth/providers/publicapi/login.go:57 +0x152
github.com/rancher/norman/api.handleAction(0xc01304dd40, 0xc006d85440, 0x0, 0x0)
   /go/pkg/mod/github.com/rancher/norman@v0.0.0-20200930000340-693d65aaffe3/api/server.go:263 +0x62
github.com/rancher/norman/api.(*Server).handle(0xc008e2e0b0, 0x4e78ba0, 0xc02df374f0, 0xc012fb5b00, 0x4ea0260, 0xc00e93ebd0, 0x4ea0260)
   /go/pkg/mod/github.com/rancher/norman@v0.0.0-20200930000340-693d65aaffe3/api/server.go:251 +0xd2
github.com/rancher/norman/api.(*Server).ServeHTTP(0xc008e2e0b0, 0x4e78ba0, 0xc02df374f0, 0xc012fb5b00)
   /go/pkg/mod/github.com/rancher/norman@v0.0.0-20200930000340-693d65aaffe3/api/server.go:180 +0x7d
github.com/gorilla/mux.(*Router).ServeHTTP(0xc001038b40, 0x4e78ba0, 0xc02df374f0, 0xc012fb5900)
   /go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212 +0xe2
github.com/rancher/rancher/pkg/multiclustermanager.(*DeferredServer).Middleware.func1(0x4e78ba0, 0xc02df374f0, 0xc012fb5900)
   /go/src/github.com/rancher/rancher/pkg/multiclustermanager/deferred.go:100 +0x98
net/http.HandlerFunc.ServeHTTP(0xc00244e1c0, 0x4e78ba0, 0xc02df374f0, 0xc012fb5900)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0010386c0, 0x4e78ba0, 0xc02df374f0, 0xc012fb5400)
   /go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212 +0xe2
github.com/rancher/rancher/pkg/api/steve/proxy.NewProxyMiddleware.func1.1(0x4e78ba0, 0xc02df374f0, 0xc012fb5400)
   /go/src/github.com/rancher/rancher/pkg/api/steve/proxy/proxy.go:75 +0x60
net/http.HandlerFunc.ServeHTTP(0xc00244e1e0, 0x4e78ba0, 0xc02df374f0, 0xc012fb5400)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/rancher/pkg/api/steve/proxy.RewriteLocalCluster.func1(0x4e78ba0, 0xc02df374f0, 0xc012fb5400)
   /go/src/github.com/rancher/rancher/pkg/api/steve/proxy/proxy.go:43 +0xef
net/http.HandlerFunc.ServeHTTP(0xc00244e200, 0x4e78ba0, 0xc02df374f0, 0xc012fb5400)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/rancher/pkg/websocket.websocketHandler.ServeHTTP(0x4e17700, 0xc00244e200, 0x4e78ba0, 0xc02df374f0, 0xc012fb5400)
   /go/src/github.com/rancher/rancher/pkg/websocket/handler.go:35 +0xa0
github.com/rancher/apiserver/pkg/middleware.ContentTypeOptions.func1(0x4e78ba0, 0xc02df374f0, 0xc012fb5400)
   /go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20201023000256-1a0a904f9197/pkg/middleware/cache.go:47 +0x131
net/http.HandlerFunc.ServeHTTP(0xc00244e220, 0x4e78ba0, 0xc02df374f0, 0xc012fb5400)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/rancher/pkg/auth/audit.auditHandler.ServeHTTP(0x4e17700, 0xc00244e220, 0xc00077aad0, 0x4e8b220, 0xc01c041b20, 0xc012fb5300)
   /go/src/github.com/rancher/rancher/pkg/auth/audit/filter.go:48 +0x227
github.com/rancher/steve/pkg/auth.ToMiddleware.func1.1(0x4e8b220, 0xc01c041b20, 0xc012fb4f00)
   /go/pkg/mod/github.com/rancher/steve@v0.0.0-20201023210011-92638937dffd/pkg/auth/filter.go:156 +0x1bd
net/http.HandlerFunc.ServeHTTP(0xc010470750, 0x4e8b220, 0xc01c041b20, 0xc012fb4f00)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/dynamiclistener/server.wrapHandler.func1(0x4e8b220, 0xc01c041b20, 0xc012fb4f00)
   /go/pkg/mod/github.com/rancher/dynamiclistener@v0.2.1-0.20200910203214-85f32491cb09/server/server.go:187 +0x95
net/http.HandlerFunc.ServeHTTP(0xc01661df80, 0x4e8b220, 0xc01c041b20, 0xc012fb4f00)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/dynamiclistener.HTTPRedirect.func1(0x4e8b220, 0xc01c041b20, 0xc012fb4f00)
   /go/pkg/mod/github.com/rancher/dynamiclistener@v0.2.1-0.20200910203214-85f32491cb09/redirect.go:20 +0x20a
net/http.HandlerFunc.ServeHTTP(0xc00e43a8c0, 0x4e8b220, 0xc01c041b20, 0xc012fb4f00)
   /usr/local/go/src/net/http/server.go:2041 +0x44
net/http.serverHandler.ServeHTTP(0xc01661b0a0, 0x4e8b220, 0xc01c041b20, 0xc012fb4f00)
   /usr/local/go/src/net/http/server.go:2836 +0xa3
net/http.(*conn).serve(0xc03c4c70e0, 0x4ea01a0, 0xc033885b80)
   /usr/local/go/src/net/http/server.go:1924 +0x86c
created by net/http.(*Server).Serve
   /usr/local/go/src/net/http/server.go:2962 +0x35c

2021/01/06 01:30:10 [ERROR] Panic serving api request:
goroutine 12252 [running]:
runtime/debug.Stack(0xc006367dc0, 0x3c26060, 0x7390960)
   /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/rancher/norman/api.(*Server).ServeHTTP.func1(0x4e78ba0, 0xc02dbf0640)
   /go/pkg/mod/github.com/rancher/norman@v0.0.0-20200930000340-693d65aaffe3/api/server.go:175 +0x7a
panic(0x3c26060, 0x7390960)
   /usr/local/go/src/runtime/panic.go:969 +0x166
github.com/rancher/rancher/pkg/auth/providers/saml.PerformSamlLogin(0x44d8204, 0x4, 0xc006fb5c20, 0x3c5dd00, 0xc02d9c0080, 0x0, 0x0)
   /go/src/github.com/rancher/rancher/pkg/auth/providers/saml/saml_provider.go:118 +0xc7
github.com/rancher/rancher/pkg/auth/providers/publicapi.(*loginHandler).createLoginToken(0xc00bec7ac0, 0xc006fb5c20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
   /go/src/github.com/rancher/rancher/pkg/auth/providers/publicapi/login.go:172 +0x905
github.com/rancher/rancher/pkg/auth/providers/publicapi.(*loginHandler).login(0xc00bec7ac0, 0xc0034a77ea, 0x5, 0xc0125a78c0, 0xc006fb5c20, 0xc00bb6d4a0, 0xc00e55b950)
   /go/src/github.com/rancher/rancher/pkg/auth/providers/publicapi/login.go:57 +0x152
github.com/rancher/norman/api.handleAction(0xc0125a78c0, 0xc006fb5c20, 0x0, 0x0)
   /go/pkg/mod/github.com/rancher/norman@v0.0.0-20200930000340-693d65aaffe3/api/server.go:263 +0x62
github.com/rancher/norman/api.(*Server).handle(0xc008e2e0b0, 0x4e78ba0, 0xc02dbf0640, 0xc012cdad00, 0x4ea0260, 0xc00e3e6360, 0x4ea0260)
   /go/pkg/mod/github.com/rancher/norman@v0.0.0-20200930000340-693d65aaffe3/api/server.go:251 +0xd2
github.com/rancher/norman/api.(*Server).ServeHTTP(0xc008e2e0b0, 0x4e78ba0, 0xc02dbf0640, 0xc012cdad00)
   /go/pkg/mod/github.com/rancher/norman@v0.0.0-20200930000340-693d65aaffe3/api/server.go:180 +0x7d
github.com/gorilla/mux.(*Router).ServeHTTP(0xc001038b40, 0x4e78ba0, 0xc02dbf0640, 0xc012cda700)
   /go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212 +0xe2
github.com/rancher/rancher/pkg/multiclustermanager.(*DeferredServer).Middleware.func1(0x4e78ba0, 0xc02dbf0640, 0xc012cda700)
   /go/src/github.com/rancher/rancher/pkg/multiclustermanager/deferred.go:100 +0x98
net/http.HandlerFunc.ServeHTTP(0xc00244e1c0, 0x4e78ba0, 0xc02dbf0640, 0xc012cda700)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0010386c0, 0x4e78ba0, 0xc02dbf0640, 0xc012cda500)
   /go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212 +0xe2
github.com/rancher/rancher/pkg/api/steve/proxy.NewProxyMiddleware.func1.1(0x4e78ba0, 0xc02dbf0640, 0xc012cda500)
   /go/src/github.com/rancher/rancher/pkg/api/steve/proxy/proxy.go:75 +0x60
net/http.HandlerFunc.ServeHTTP(0xc00244e1e0, 0x4e78ba0, 0xc02dbf0640, 0xc012cda500)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/rancher/pkg/api/steve/proxy.RewriteLocalCluster.func1(0x4e78ba0, 0xc02dbf0640, 0xc012cda500)
   /go/src/github.com/rancher/rancher/pkg/api/steve/proxy/proxy.go:43 +0xef
net/http.HandlerFunc.ServeHTTP(0xc00244e200, 0x4e78ba0, 0xc02dbf0640, 0xc012cda500)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/rancher/pkg/websocket.websocketHandler.ServeHTTP(0x4e17700, 0xc00244e200, 0x4e78ba0, 0xc02dbf0640, 0xc012cda500)
   /go/src/github.com/rancher/rancher/pkg/websocket/handler.go:35 +0xa0
github.com/rancher/apiserver/pkg/middleware.ContentTypeOptions.func1(0x4e78ba0, 0xc02dbf0640, 0xc012cda500)
   /go/pkg/mod/github.com/rancher/apiserver@v0.0.0-20201023000256-1a0a904f9197/pkg/middleware/cache.go:47 +0x131
net/http.HandlerFunc.ServeHTTP(0xc00244e220, 0x4e78ba0, 0xc02dbf0640, 0xc012cda500)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/rancher/pkg/auth/audit.auditHandler.ServeHTTP(0x4e17700, 0xc00244e220, 0xc00077aad0, 0x4e8b220, 0xc02254d7a0, 0xc012cda400)
   /go/src/github.com/rancher/rancher/pkg/auth/audit/filter.go:48 +0x227
github.com/rancher/steve/pkg/auth.ToMiddleware.func1.1(0x4e8b220, 0xc02254d7a0, 0xc01343f500)
   /go/pkg/mod/github.com/rancher/steve@v0.0.0-20201023210011-92638937dffd/pkg/auth/filter.go:156 +0x1bd
net/http.HandlerFunc.ServeHTTP(0xc010470750, 0x4e8b220, 0xc02254d7a0, 0xc01343f500)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/dynamiclistener/server.wrapHandler.func1(0x4e8b220, 0xc02254d7a0, 0xc01343f500)
   /go/pkg/mod/github.com/rancher/dynamiclistener@v0.2.1-0.20200910203214-85f32491cb09/server/server.go:187 +0x95
net/http.HandlerFunc.ServeHTTP(0xc01661df80, 0x4e8b220, 0xc02254d7a0, 0xc01343f500)
   /usr/local/go/src/net/http/server.go:2041 +0x44
github.com/rancher/dynamiclistener.HTTPRedirect.func1(0x4e8b220, 0xc02254d7a0, 0xc01343f500)
   /go/pkg/mod/github.com/rancher/dynamiclistener@v0.2.1-0.20200910203214-85f32491cb09/redirect.go:20 +0x20a
net/http.HandlerFunc.ServeHTTP(0xc00e43a8c0, 0x4e8b220, 0xc02254d7a0, 0xc01343f500)
   /usr/local/go/src/net/http/server.go:2041 +0x44
net/http.serverHandler.ServeHTTP(0xc01661b0a0, 0x4e8b220, 0xc02254d7a0, 0xc01343f500)
   /usr/local/go/src/net/http/server.go:2836 +0xa3
net/http.(*conn).serve(0xc00f0b2e60, 0x4ea01a0, 0xc00f095580)
   /usr/local/go/src/net/http/server.go:1924 +0x86c
created by net/http.(*Server).Serve
   /usr/local/go/src/net/http/server.go:2962 +0x35c
```

**Other details that may be helpful:**
After re-attempting Okta integration (repasted keys into text box) it worked.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.5.3

- Installation option (single install/HA):
HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
local
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
aws m5a.large - 2 vCPU, 8GB RAM
- Kubernetes version (use `kubectl version`):

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T21:51:49Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.12+k3s2", GitCommit:"1f3505370b8067fc90ab627697053fc8c5c133ac", GitTreeState:"clean", BuildDate:"2020-12-07T21:08:08Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):
n/a, containerd

#gz14086

gz#14541

**Comment by izaac** [comment]
I've not been able to repro this in `v2.5.5` or `master/2.5-head` as I'm hitting this other issue:

https://github.com/rancher/rancher/issues/31163

**Comment by izaac** [comment]
I was able to find a workaround to enable the auths in an 2.5.5 HA setup.
https://github.com/rancher/rancher/issues/31163#issuecomment-775382146

But I'm still on the lookout for the Panic

**Comment by izaac** [comment]
Rancher `v2.5.5` 

i've tried to reproduce this on 2.5.5 using keycloak and okta but I didn't get the panic in the rancher logs (loglevel was set to `TRACE`).

I didn't see the panic when enabling the Auth where I got a 404 sometimes on HA but this is followed in a separate issue: https://github.com/rancher/rancher/issues/31163

[According to dev](https://github.com/rancher/rancher/pull/31079#discussion_r568866276) this happens on Login, so I also tried using a couple of accounts with Keycloak and Okta but I didn't get the panic reported here. 

Are there any other configuration like special RBAC or site access that I could try for reproducing this? @dramich
